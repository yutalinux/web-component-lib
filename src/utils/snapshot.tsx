import { type Attributes, type FunctionComponent, createElement } from "react";
import { renderToString } from "react-dom/server";

export function snapshotElement<T extends object>(
  type: FunctionComponent<T>,
  args: (Attributes & T) | null,
) {
  const element = createElement(type, args);
  return renderToString(element);
}
