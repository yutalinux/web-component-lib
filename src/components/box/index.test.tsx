import { snapshotElement } from "@/utils/tests/snapshot";
import { expect, test } from "vitest";
import Box from ".";

test("Box Snapshot", () => {
  expect(snapshotElement(Box, { orientation: "vertical" })).toMatchSnapshot();
  expect(snapshotElement(Box, { orientation: "horizontal" })).toMatchSnapshot();
});
