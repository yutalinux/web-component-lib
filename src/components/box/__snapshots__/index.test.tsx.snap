// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Snapshot 1`] = `"<div class="flex gap-2 flex-col"></div>"`;

exports[`Box Snapshot 2`] = `"<div class="flex gap-2"></div>"`;
